### DISCUSSION TITLE
Failed to read file for Multiscale Tesselation

### DISCUSSION THREAD

**Discussion by danishTUE** [neperfepx/neper#912] [Q&A] [answered]
Hi Romain,

I am trying to generate a 3-level multiscale tesselation with lamellar morphology as the lowest scale. I use the fixed number of lamellae option as suggested here : [https://github.com/neperfepx/neper/discussions/901](url). I am defining the normals of the lamellae using a file. However, the script ends up with the error: "Failed to read file". Taking a cue from the discussion here: [https://github.com/neperfepx/neper/discussions/304](url), I tried with creating the lamnormal file with the Linux text editor and also used dos2unix to convert it into unix format, if in case that's the issue. However, this error persists. Below is my command:
 
```
neper -T -n from_morpho::4::6 -id 1 -dim 3 -domain 'cube(1,1,1)' -morpho "cube(1)::voronoi::lamellar(v=file(lamnormal.txt))" -format 'vtk','tess' -o 5_Cube_Voro_Lam
```

And the generated output is as follows:
```
========================    N   e   p   e   r    =======================
Info   : A software package for polycrystal generation and meshing.
Info   : Version 4.9.1-4
Info   : Built with: gsl|muparser|opengjk|openmp|nlopt|libscotch (full)
Info   : Running on 12 threads.
Info   : <https://neper.info>
Info   : Copyright (C) 2003-2022, and GNU GPL'd, by Romain Quey.
Info   : No initialization file found (`/home/danish/.neperrc').
Info   : ---------------------------------------------------------------
Info   : MODULE  -T loaded with arguments:
Info   : [ini file] (none)
Info   : [com line] -n from_morpho::4::6 -id 1 -dim 3 -domain cube(1,1,1)
         -morpho cube(1)::voronoi::lamellar(v=file(lamnormal.txt))
         -format vtk,tess -o 5_Cube_Voro_Lam
Info   : ---------------------------------------------------------------
Info   : Reading input data...
Info   : Creating domain...
Info   : Creating tessellations...
Info   :   [level] Running in multiscale mode (3 scales)...
Info   :   [    1]  1/1 Info   :   - Running tessellation...
Info   : Generating crystal orientations...
Info   :   [    2]  1/1 Info   :   - Setting seeds...
Info   :   - Running tessellation...
Info   : Generating crystal orientations...
Info   :   [    3]  1/4 Info   :   - Running tessellation...
Error  : Failed to read file `file(lamnormal.txt)'.
./5_Cube_Voro_Lam_LamNorm.sh: line 8:  1054 Aborted                 (core dumped) neper -T -n from_morpho::4::6 -id 1 -dim 3 -domain 'cube(1,1,1)' -morpho "cube(1)::voronoi::lamellar(v=file(lamnormal.txt))" -format 'vtk','tess' -o 5_Cube_Voro_Lam
```

I am attaching the lamnormals.txt file as well. Can you kindly suggest what may be the issue here. 

Many thanks in advance.

Regards,

Danish

[lamnormal.txt](https://github.com/user-attachments/files/17217770/lamnormal.txt)

**Comment by rquey** [comment, marked as answer]
You have to format `lamwidth.txt` as a [multiscale cell file](https://neper.info/doc/fileformat.html#multiscale-cell-file). Since you are specifying information at scale 3, the "ids" on the first column are multiscale cell ids formattted as `<scale1_parent_cell>::<scale2_parent_cell>`. In your case, replace
```
1
2
3
4
```
by
```
1::1
1::2
1::3
1::4
```

If you had two cells at *first* scale, each containing 4 secondary cells, the ids would be
```
1::1
1::2
1::3
1::4
2::1
2::2
2::3
2::4
```
